GoogleChromeLabs/ps-analysis-tool
# Feature: Add global filtering capabilities

## Description
This PR adds global filtering capabilities to the landing/insights page of the extension, site, and sitemap report.
- The report downloaded will have a header showing applied filters and filtered matrix count.
- Whenever a filter is applied on the landing page it should update the count of matrices and filters should persist across tables.


## Relevant Technical Choices



## Testing Instructions
- Open the PSAT panel and navigate to the landing page.
- Select the filters and they should be applied to the matrix, and chips should be available.
- Now navigate to the table, the filters will be present with filtered data.
- Now download the report, it should also contain chips and filtered data.
- Now try in cli with available flags.


## Additional Information:



## Screenshot/Screencast


<img width="1470" alt="Screenshot 2024-07-09 at 17 03 21" src="https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/3b186ece-abeb-4124-8ee0-b3d0db44d66c">




---

## Checklist






Fixes #750
